### PR TITLE
Introduce CSpell linter

### PR DESCRIPTION
**Pull request by javydekoning**
# Why?

- Introducing CSpell to reduce the number of spelling mistakes in our code. 
- Also a first effort to reduce non inclusive language. 
- Standardize on en_US instead (we are mixing en_US with en_UK). 

Describe why you are proposing these changes

*Issue #, if available:*

For example, to avoid typo's such as: 
https://github.com/awslabs/aws-deployment-framework/pull/567

## What?

Description of changes:

- Added a custom dictionary under `/linters`
- Added (and removed from repo) blocked words: 'whitelist' and 'blacklist'
- Fixed spelling mistakes such as: 'TERIMIATION', 'MANGEMENT', 'DOESNT'.
- Enabled the linter in our megalinter config. 
- Changed to en_US. For example: 'Standardised -> Standardized', 'Initialisation -> Initialization'

## What is NOT covered. 

Changing 'Master' to 'Management'. This is still used at many places in this repository and changing this will have an impact on current installs. Best to tackle that in the next major version of ADF.

---

By submitting this pull request, I confirm that you can use, modify, copy, and
redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by sbkok — CHANGES_REQUESTED** [review]
Thanks for adding this so quickly!
This well help a lot with reviewing code and ensuring typos don't accidentally end up in logical resource names as you mentioned.

I also agree on the inclusive wording. I worked on a branch a while ago to change the master account references into management account (in the code mainly), but I parked that until we can start working on the next major version.

File: linters/custom-adf-dict.txt [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+oxsecurity
+pipelinenoti
+pygtk
+pylintrc
+rcfile
+releasever
+rexec
+runas
+sdkman
+stefanzweifel
```
[the remark this answers is not in the document]
Comment (javydekoning) [reply linters/custom-adf-dict.txt:46]: [In the .github workflow](https://github.com/awslabs/aws-deployment-framework/blob/master/.github/workflows/mega-linter.yml#L68)

File: src/lambda_codebase/initial_commit/bootstrap_repository/adf-bootstrap/deployment/lambda_codebase/pipeline_management/create_or_update_rule.py [hunk 2]
```diff
@@ -32,7 +32,7 @@ def lambda_handler(pipeline, _):
     if not _cache:
         _cache = Cache()
         METRICS.put_metric_data(
-            {"MetricName": "CacheInitalised", "Value": 1, "Unit": "Count"}
+            {"MetricName": "CacheInitalized", "Value": 1, "Unit": "Count"}
```
[the remark this answers is not in the document]
Comment (javydekoning) [reply src/lambda_codebase/initial_commit/bootstrap_repository/adf-bootstrap/deployment/lambda_codebase/pipeline_management/create_or_update_rule.py:35]: Interesting, even my local CSpell plugin doesn't catch this typo....

**Review by StewartW — DISMISSED** [review]
Looks good. Great addition. Ironic spelling mistake in the PR title is intentional yeah ;)

**Review by sbkok — APPROVED** [review]

**Review by StewartW — APPROVED** [review]

**Pull request merged**